#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 955f550d5e769433eb57143e784851b7698a55ba Mon Sep 17 00:00:00 2001
From: Markus Montkowski <markus.montkowski@gnupg.com>
Date: Thu, 16 Oct 2025 12:28:08 +0200
Subject: [PATCH] Opened/Closed encrypted pgp messages unencrypted with read as
 plain

* src/mail.cpp: put empty string into Body prior to passing on the close

--
clear the body to prevent plain text leak
GnuPG-Bug-Id: 7858
---
 src/mail.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/mail.cpp b/src/mail.cpp
index cc10cfe..26bef16 100644
--- a/src/mail.cpp
+++ b/src/mail.cpp
@@ -3031,9 +3031,12 @@ Mail::close (bool restoreSMIMEClass)
         }
       else
         {
-           setPassWrite (true);
-           log_debug ("%s:%s: Close successful. Next write may pass.",
-                      SRCNAME, __func__);
+          // Just to be sure if body is falsely reported as empty as in T7857
+          // put an empty string in it to prevent plain text leak
+          put_oom_string (m_mailitem, "Body", "" );
+          setPassWrite (true);
+          log_debug ("%s:%s: Close successful. Next write may pass.",
+                     SRCNAME, __func__);
         }
       gpgol_release (attachments);
       xfree (body);
--
2.39.2
